modules/tinydnsbackend/tinydnsbackend.cc

```diff
@@ -177,9 +179,14 @@ void TinyDNSBackend::getAllDomains(vector<DomainInfo> *domains, bool include_dis
       di.notified_serial = sd.serial;
       di.kind = DomainInfo::Master;
       di.last_check = time(0);
-      domains->push_back(di);
+      if (result.find(di) != result.end())
+        continue;
+      result.insert(di);
```

Comment:
You could always insert instead of looking up first, by the way ;-)

Reply:
what happens if i do that though

Reply:
The insertion will be prevented and the existing value will remain in the set.

Reply:
huh. I would've expected exception at least.